auth0/v3/test/management/test_rest.py

```diff
@@ -1,12 +1,45 @@
-import unittest
 import json
+import unittest
+
 import mock
-from ...management.rest import RestClient
+
 from ...exceptions import Auth0Error
+from ...management.rest import RestClient, RateLimiter
 
 
-class TestRest(unittest.TestCase):
+class TestRatelimiter(unittest.TestCase):
+    def test_update(self):
+        rl = RateLimiter()
+
+        rl.update({'x-ratelimit-remaining': '5', 'x-ratelimit-reset': '2'})
+
+        self.assertEqual(rl.ratelimit_remaining, 5)
+        self.assertEqual(rl.ratelimit_reset, 2)
+
+    @mock.patch('time.sleep')
+    def test_ensure_limit_without_sleep(self, mock_sleep):
+        rl = RateLimiter()
+        rl.update({'x-ratelimit-remaining': '1', 'x-ratelimit-reset': '2'})
 
+        rl.ensure_limit()
+
+        mock_sleep.assert_not_called()
+
+    @mock.patch('time.time')
+    @mock.patch('time.sleep')
+    def test_ensure_limit_without_sleep(self, mock_sleep, mock_time):
+        current_time = 1342342
+        mock_time.return_value = current_time
+        rl = RateLimiter()
+
+        rl.update({'x-ratelimit-remaining': '0', 'x-ratelimit-reset': str(current_time + 2)})
+
+        rl.ensure_limit()
+
+        mock_sleep.assert_called_with(2)
```

Comment:
why is this test case called "without sleep" when you want to assert that `sleep` got called?